src/sample-executors/BaseExecutor.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {Owned} from "solmate/src/auth/Owned.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+
+import {ResolvedOrder, SignedOrder} from "../base/ReactorStructs.sol";
+import {IReactorCallback} from "../interfaces/IReactorCallback.sol";
+import {IReactor} from "../interfaces/IReactor.sol";
+import {Multicall} from "./Multicall.sol";
+
+struct PermitData {
+    address token;
+    bytes data;
+}
+
+abstract contract BaseExecutor is IReactorCallback, Multicall, Owned {
+    IReactor public immutable reactor;
+
+    constructor(IReactor _reactor, address _owner) Owned(_owner) {
+        reactor = _reactor;
+    }
+
+    function reactorCallback(ResolvedOrder[] calldata, bytes calldata callbackData) external virtual;
+
+    function execute(SignedOrder memory order, bytes memory callbackData) external payable {
```

Comment:
should these be onlyOwner?

Reply:
mmm i thought we wanted execute to be gated by onlyWhitelstedCaller?